lib/Partitioner/Partitioner.cpp

```diff
@@ -569,68 +581,390 @@ void Partitioner::doPartitioning(Function *F, NodeToFunctionMap &mapping) {
   }
 }
 
+void Partitioner::doFuncListPartitioning(llvm::StringRef funcName,
+                                         std::vector<Function *> funcs,
+                                         NodeToFunctionMap &mapping) {
+  // The dummy node.
```

Comment:
What is a dummy node?  I don't understand the algorithm. It should be something like:

// Add a dummy node that will allow us to do XXX ...